tests/lang_stmts/casestmt/tbranch_with_empty_set.nim

```diff
@@ -0,0 +1,22 @@
+discard """
+  description: '''
+    Regression test for where an empty set/array value in a label position
+    counted towards the number of required labels, leading to errors being
+    reported for valid case statements.
+  '''
+  action: compile
+"""
+
+var x: bool
+
+case x
+of true, []: # would cause "not all cases are covered"
```

Comment:
Huh, I didn't know an array of values was legal.

Reply:
Yeah, it's not really documented anywhere, I think. `semCaseBranch` uses type-based dispatching now, so it's a least made somewhat more obvious in the code.